internal/configs/test_file.go

```diff
@@ -701,12 +794,36 @@ func decodeTestRunBlock(block *hcl.Block, file *TestFile) (*TestRun, hcl.Diagnos
 		rawDiags := gohcl.DecodeExpression(attr.Expr, nil, &r.StateKey)
 		diags = append(diags, rawDiags...)
 	}
+	// If there's no user-supplied state_key value, set it based on config
+	// under test.
+	if r.StateKey == "" {
+		if r.ConfigUnderTest != nil {
+			// If the run is using an alternate module under test, the source of
+			// that module is returned as the state key.
+			r.StateKey = r.Module.Source.String()
+		} else {
+			// Otherwise, an empty string is returned, and this denotes that the
+			// run is using the root module under test.
+			r.StateKey = ""
```

Comment:
nit: this is redundant, the state key is already empty

Reply:
Thanks! 🤦🏻  I've left the comment in place but removed the unnecessary assignment: [54dd772](https://github.com/hashicorp/terraform/pull/36541/commits/54dd772104a61c2500b8826e223ceed7cadd1ad7)

Reply:
We could potentially change the state key for the root module to stop being an empty string, but I imagine that would add more validation issues (i.e stop name collision with a user-set state key)